tutorials/BalanceLaws/tendency_specification_layer.jl

```diff
@@ -0,0 +1,129 @@
+# # A functional tendency specification layer
+
+# In the balance law (mutating) functions, where we specify fluxes and sources,
+
+# - `flux_first_order!`
+# - `flux_second_order!`
+# - and `source!`,
+
+# an additional (functional) tendency specification
+# layer can be placed on-top that has several nice
+# properties. The functional layer:
+
+# - Separates tendency definitions from which tendencies are included in a particular model.
+# - Reduces duplicate implementations of tendency definitions (e.g., in optional submodel variants)
+# - Allows a more flexible combination of tendencies
+# - Allows a simple way to loop over all tendencies for all prognostic variables and recover
+#   _each_ flux / source term. This will allow us a simple way to evaluate, for example, the energy budget.
+
+# ## Preliminaries
+
+# Make running locally easier from ClimateMachine.jl/:
+if !("." in LOAD_PATH)
+    push!(LOAD_PATH, ".")
+end
+
+# First, using necessary modules:
+using ClimateMachine.BalanceLaws
+using ClimateMachine.VariableTemplates
+using StaticArrays, Test
+
+# Import methods for building blocks
+import ClimateMachine.BalanceLaws: prognostic_vars, eq_tends
+# Import methods for adding tendency specification layer
+import ClimateMachine.BalanceLaws: flux
+
+# Define a balance law
+struct MyBalanceLaw <: BalanceLaw end
+
+# ## Building blocks
+
+# Define prognostic variables:
+struct X <: PrognosticVariable end
+struct Y <: PrognosticVariable end
+
+# Define `prognostic_vars`, which returns _all_ prognostic vars
+prognostic_vars(::MyBalanceLaw) = (X(), Y());
+
+# Define some tendencies:
+
+# Tendency definitions can be shared across different
+# prognostic variables:
+struct F1{PV} <: TendencyDef{Flux{FirstOrder}, PV} end
+
+# We can also permit tendency definitions to be defined for
+# only the relevant prognostic variables: here, we restrict
+# the second order flux FY2 to be defined only for Y.
+struct FY2{PV <: Y} <: TendencyDef{Flux{SecondOrder}, PV} end
+
+struct S1{PV} <: TendencyDef{Source, PV} end
+struct S2{PV} <: TendencyDef{Source, PV} end
+
+# Define `eq_tends` given
```

Comment:
Expand a little on what `eq_tends` does? "Define tendencies for the balance law..." or something?

Reply:
I added more description of what this does..